python/grass/gunittest/loader.py

```diff
@@ -117,6 +118,7 @@ def discover_modules(
             full = os.path.join(root, testsuite_dir)
 
             all_files = os.listdir(full)
+            files = [f for f in all_files]
             if file_pattern:
                 files = fnmatch.filter(all_files, file_pattern)
             if file_regexp:
```

Comment:
This change perhaps reveals a flaw in the original code. The whole block looks like it should be series of filters, so first files == all_files, but then fnmatch is applied to files, and then regexp, and finally exclude.

Reply:
I tried that change initially, but didn't keep it for my final version, as it wasn't in scope for the PR, and didn't manage to understand if it was supposed to be exclusive or additive. So I opted to not change the existing behaviour at all, for the time left for gunittest. That change only made sure that if exclude was passed, it would be used.

One of the two calls of discover modules doesn't pass exclude, and I still have difficulty understanding in which order are the classes called.

Reply:
If you can assure the intended behavior and consequences, I had a commit that looked like this:

https://github.com/echoix/grass/pull/220/commits/93637eeda8520cd7671b4af0a388b47934d20379
![image](https://github.com/user-attachments/assets/0de8bce1-0adf-410b-94c8-7bb5414462dd)

In that case, `all_files` could be changed to only `files`, and adapt the 2 usages of `all_files`

Reply:
Given that this is how exclude already works, I'm reasonably confident that filtering through `files` will work. It also looks like the most consistent behavior.